src/core/controllers/base.go

```diff
@@ -108,8 +108,46 @@ func (cc *CommonController) Login() {
 	cc.PopulateUserSession(*user)
 }
 
-// LogOut Habor UI
+// LogOut Harbor UI
 func (cc *CommonController) LogOut() {
+	// redirect for OIDC logout, excludes the admin user.
+	securityCtx, ok := security.FromContext(cc.Context())
+	if !ok {
+		log.Error("Failed to get security context")
+		cc.CustomAbort(http.StatusInternalServerError, "Internal error.")
+	}
+	principal := securityCtx.GetUsername()
+	if principal != "" {
+		if redirectForOIDC(cc.Ctx.Request.Context(), principal) {
+			u, err := user.Ctl.GetByName(cc.Context(), principal)
+			if err != nil {
+				log.Errorf("Failed to get user by name: %s, error: %v", principal, err)
+				cc.CustomAbort(http.StatusInternalServerError, "Internal error.")
```

Comment:
same here, in the error message we have better error message why a generic in the reponse?

Reply:
ditto